dali/kernels/audio/mel_scale/mel_filter_bank_gpu_test.cc

```diff
@@ -0,0 +1,183 @@
+    std::mt19937 rng;
+    UniformRandomFill(in_.cpu(), rng, 0.0, 1.0);
+  }
+  TensorListShape<4> data_shape_;
+  int nfilter_ = 4;
+  float sample_rate_ = 16000, freq_low_ = 0, freq_high_ = 8000;
+  TestTensorList<float, 4> in_;
+  int64_t axis_;
+};
+
+TEST_P(MelScaleGpuTest, MelScaleGpuTest) {
+  using T = float;
+  HtkMelScale<float> mel_scale;
+  constexpr int Dims = 4;
+  auto shape = data_shape_;
+  auto batch_size = data_shape_.num_samples();
+  int nfft = (shape[0][axis_] - 1) * 2;
+  std::vector<int> nwin(batch_size, 1);
+  if (axis_ < Dims - 1) {
+    for (int i = 0; i < batch_size; ++i) {
+        auto sh = shape[i];
+        nwin[i] = volume(sh.begin() + axis_ + 1, sh.end());
+    }
+  }
+  std::vector<int> nframes(batch_size, 1);
+  if (axis_ > 0) {
+    for (int i = 0; i < batch_size; ++i) {
+      auto sh = shape[i];
+      nframes[i] = volume(sh.begin(), sh.begin() + axis_);
+    }
+  }
+
+  TensorListShape<4> out_shape(batch_size);
+  for (int i = 0; i < batch_size; ++i) {
+    auto s = data_shape_[i];
+    s[axis_] = nfilter_;
+    out_shape.set_tensor_shape(i, s);
+  }
+  std::vector<T> out_sizes;
+  for (int i = 0; i < batch_size; ++i) {
+    out_sizes.push_back(volume(out_shape[i]));
+  }
+
+  T mel_low = mel_scale.hz_to_mel(freq_low_);
+  T mel_high = mel_scale.hz_to_mel(freq_high_);
+
+  T mel_delta = (mel_high - mel_low) / (nfilter_ + 1);
+  T mel = mel_low;
+  LOG_LINE << "Mel frequency grid (Hz):";
+  for (int i = 0; i < nfilter_+1; i++, mel += mel_delta) {
+    LOG_LINE << " " << mel_scale.mel_to_hz(mel);
+  }
+  LOG_LINE << " " << mel_scale.mel_to_hz(mel_high) << "\n";
+
+  LOG_LINE << "FFT bin frequencies (Hz):";
+  for (int k = 0; k < nfft / 2 + 1; k++) {
+    LOG_LINE << " " << (k * sample_rate_ / nfft);
+  }
+  LOG_LINE << "\n";
+
+  auto fbanks = ReferenceFilterBanks(nfilter_, nfft, sample_rate_, freq_low_, freq_high_);
+  std::vector<std::vector<float>> expected_out;
+  for (int i = 0; i < batch_size; ++i) {
+    expected_out.emplace_back(out_sizes[i], 0.f);
+  }
+
+  for (int b = 0; b < batch_size; ++b) {
+    auto in_view = in_.cpu()[b];
+    for (int s = 0; s < nframes[b]; ++s) {
+      for (int j = 0; j < nfilter_; j++) {
+        for (int t = 0; t < nwin[b]; t++) {
+          auto &out_val = expected_out[b][(s * nfilter_ + j) * nwin[b] + t];
+          for (int i = 0; i < nfft/2+1; i++) {
+            out_val += fbanks[j][i] * in_view.data[(s * (nfft/2+1) + i) * nwin[b] + t];
+          }
+        }
+      }
+    }
+  }
+
+  KernelContext ctx;
+  ctx.gpu.stream = 0;
+  kernels::audio::MelFilterBankArgs args;
+  kernels::KernelManager kmgr;
+  args.axis = axis_;
+  args.nfft = nfft;
+  args.nfilter = nfilter_;
+  args.sample_rate = sample_rate_;
+  args.freq_low = freq_low_;
+  args.freq_high = freq_high_;
+  args.mel_formula = MelScaleFormula::HTK;
+  args.normalize = false;
+
+  using Kernel = kernels::audio::MelFilterBankGpu<T, Dims>;
+  kmgr.Initialize<Kernel>();
+  kmgr.Resize<Kernel>(1, 1);
```

Comment:
I would explicitly set stream to the kernel manager here (just in case).

Reply:
? Kernel manager doens't need streams. KernelContext does.

Reply:
My bad.